assets/src/components/detours/detourDropdown.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react"
+import { DropdownItem, DropdownMenu } from "../map/dropdown"
+import { Route, RoutePattern, Shape } from "../../schedule"
+import { Popup } from "react-leaflet"
+import { PointTuple } from "leaflet"
+import { LatLngLiteral } from "leaflet"
+
+export interface StartDetourProps {
```

Comment:
@mbta/skate-developers I feel like these types have become kind of a mess. I feel like `StartDetourProps` (or something like it) should exist (because otherwise we'll end up prop-drilling like a million separate params), but I'm not sure it should live here.

Reply:
I think you should store the route pattern id, and rederive the route name, description, origin, direction, and shape in the `detourModal`.

Reply:
👍 Is there a hook already that will let me exchange a route pattern ID for all of that info?

If not, feels like a good idea to make that hook, huh

Reply:
Oooorrrr... I could pass the route pattern itself as a prop?

Reply:
there is `usePatternsByIdForRoute` but there isn't anything that directly goes from `routePatternId` to `RoutePattern` it seems. I agree that having that would be a great addition.

Reply:
I think it's fine if the `routeId` is needed in addition to the `routePatternId` to satisfy `usePatternsByIdForRoute`

Reply:
Hmmmmm I'm becoming a little less sold on this for a few reasons:

1. I don't love the idea of re-deriving information that we already know.
2. `StartDetourProps` has all of its fields as not-null - if any of them are `null`, then the `DetourDropdown` simply doesn't appear. I'd rather not do a bunch of null-checking on fields that we already know in-context are not null.

I'm leaning a bit more towards having `StartDetourProps` move into `detour.ts` and be like... `OriginalRoutePattern` or something like that, while keeping its existing structure, and then instead of passing around each of the fields separately, passing them together.

Reply:
> I don't love the idea of re-deriving information that we already know.

The problem I have is assuming that we're storing information that'll never update, whereas pulling this information from hooks via ID's allows us to centralize the way we access this data (allowing us to do more refactors and caching that we've had queued up)

Maybe this data shouldn't change, but it's already an established pattern within `state.ts` that we store Id's, not objects. There are a few exceptions to this, but I'd prefer to get rid of those as well.

---

All in all though, I'm fine with this being non-blocking and you continuing with what you think is best, as we can refactor later if it's a problem.

Reply:
Kind of a big diff, but not a huge change as it turns out.

[This](https://github.com/mbta/skate/pull/2377/commits/9d600e4108d7ac49cdd64c8147da8939529d14e7) is what I landed on.

- Renamed `StartDetourProps` to `OriginalRoute`.
- Moved `OriginalRoute` into `detour.ts` because it's shared around and is a pretty core part of the detour-editing world.
- Wherever _all_ of the fields of `OriginalRoute` were being passed around, replaced the collection of fields with just one field of type `OriginalRoute` (the fields are still passed "by hand" into `DetourMap` and `DiversionPanel`, but that's a problem for another day)

Reply:
(And then [this](https://github.com/mbta/skate/pull/2377/commits/4e701387fb08ffd236589e02f124a3e9ac5cdaae) to fix the tests and other compilation issues)